src/main/java/Duke.java

```diff
@@ -1,10 +1,202 @@
+        return (isTwoWordInput && isFirstWordMark && isSecondWordNumber);
+    }
+
+    public static boolean isUnmark(String[] arrayOfInput) {
+        // if input length (separated by " " is 2) && first word == "unmark" && second word only contains numbers
+        boolean isTwoWordInput = arrayOfInput.length == 2;
+        boolean isFirstWordUnmark = arrayOfInput[0].equals("unmark");
+        boolean isSecondWordNumber = isStringInteger(arrayOfInput[1]);
+        return (isTwoWordInput &&isFirstWordUnmark && isSecondWordNumber);
+    }
+
+    public static boolean isStringInteger(String input) {
+        // takes in a string and checks whether the string only contains digits
+        char[] inputInArray = input.toCharArray();
+        for (char c : inputInArray) {
+            if (!Character.isDigit(c)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static void markTask(Task[] tasks, String[] arrayOfInput) {
+        tasks[Integer.parseInt(arrayOfInput[1]) - 1].markAsDone();
+        printMarkOrUnmarkStatement(tasks, Integer.parseInt(arrayOfInput[1]) - 1);
+    }
+
+    public static void unmarkTask(Task[] tasks, String[] arrayOfInput) {
+        tasks[Integer.parseInt(arrayOfInput[1]) - 1].markAsNotDone();
+        printMarkOrUnmarkStatement(tasks, Integer.parseInt(arrayOfInput[1]) - 1);
+    }
+
+    public static void printMarkOrUnmarkStatement(Task[] tasks, int taskIndex) {
+        System.out.println(SEGMENT_LINE);
+        if (tasks[taskIndex].isDone) {
+            System.out.println(" Awesome! I've marked this task as done:");
+        } else {
+            System.out.println(" OK, I've marked this task as not done yet:");
+        }
+        System.out.println("   " + tasks[taskIndex].getFullTaskDetail());
+        System.out.println(SEGMENT_LINE + System.lineSeparator());
+    }
+
+    public static void addTask(String input, Task[] tasks, String[] arrayOfInput) {
+        boolean isInputTodo = arrayOfInput[0].equals("todo");
+        boolean isInputDeadline = arrayOfInput[0].equals("deadline");
+        boolean isInputEvent = arrayOfInput[0].equals("event");
+        if (isInputTodo) {
+            addTodoTask(tasks, input);
+        } else if (isInputDeadline) {
+            addDeadlineTask(tasks, arrayOfInput);
+        } else if (isInputEvent) {
+            addEventTask(tasks, arrayOfInput);
+        } else {
+            // if input is just the task itself without keyword
+            tasks[Task.totalTasks] = new Task(input);
+        }
+        echoAddTasks(tasks);
+        Task.incrementTotalTasks();
+    }
+
+    public static void addTodoTask(Task[] tasks, String input) {
+        String[] commandTaskNameArray = input.split(" ", 2);
+        tasks[Task.totalTasks] = new Todo(commandTaskNameArray[1]);
+    }
+
+    public static void addDeadlineTask(Task[] tasks, String[] arrayOfInput) {
+        String taskName = "";
```

Comment:
This looks like a C implementation (by prof)
I think can use string.split to format inputs instead.